docs/source/tutorials/token_classification/utils.py

```diff
@@ -0,0 +1,242 @@
+                        top=10, exclude=[], verbose=True): 
+    n = pred_probs[0].shape[1] 
+    count = {} 
+    for issue in issues: 
+        i, j = issue 
+        word = given_words[i][j] 
+        label = labels[i][j] 
+        pred = pred_probs[i][j].argmax() 
+        if word not in count: 
+            count[word] = np.zeros([n, n], dtype=int) 
+        if (label, pred) not in exclude: 
+            count[word][label][pred] += 1 
+    words = [word for word in count.keys()] 
+    n = count[words[0]].shape[0] 
+    freq = [np.sum(count[word]) for word in words] 
+    rank = np.argsort(freq)[::-1][:top] 
+    
+    for r in rank: 
+        word = words[r] 
+        matrix = count[word] 
+        most_frequent = np.argsort(matrix.flatten())[::-1] 
+        print("'%s' is mislabeled %d times" % (word, freq[r])) 
+        if verbose: 
+            print('-----------------------------') 
+            for f in most_frequent: 
+                i, j = f // n, f % n 
+                if matrix[i][j] == 0: 
+                    break 
+                print('labeled as %s, but predicted as %s %d times' % 
+                      (entities[i], entities[j], matrix[i][j])) 
+        print() 
+        
+def search_token(token, issues, labels, pred_probs, given_words, sentences, entities): 
+    d = {} 
+    for issue in issues: 
+        i, j = issue 
+        if given_words[i][j].lower() == token.lower(): 
+            if i not in d.keys(): 
+                d[i] = [] 
+            given = entities[labels[i][j]] 
+            pred = entities[pred_probs[i][j].argmax()] 
+            d[i].append((j, given, pred)) 
+    indicies = list(d.keys()) 
+    indicies.sort() 
+    
+    def get_mapping(sentence, tokens): 
+        i, indicies = 0, [] 
+        for token in tokens: 
+            indicies.append(sentence[i:].index(token) + i) 
+            i += len(token) 
+        return indicies 
+    
+    for index in indicies: 
+        mapping = get_mapping(sentences[index], given_words[index]) 
+        info = sorted(d[index], key=lambda x: x[0]) 
+        mapped_indicies = [mapping[j] for j, _, _ in info] 
+        s = 'Sentence %d: ' % index 
+        i = 0 
+        for mapped_index in mapped_indicies: 
+            s += sentences[index][i:mapped_index] 
+            s += colored(token, 'red') 
+            i = mapped_index + len(token) 
+        s += sentences[index][i:] 
+        print(s) 
+        for _, given, pred in info: 
+            print('Given label: %s, predicted label: %s\n' % (given, pred)) 
+    return d 
+
+def show_sentence_issues(scores, token_scores, pred_probs, given_words, sentences, given_labels, entities, top=10): 
+    ranking = np.argsort(scores) 
+    for r in ranking[:top]: 
+        print('Sentence %d: score=%.6f' % (r, scores[r])) 
+        issue_index = np.argmin(token_scores[r]) 
+        issue_word = given_words[r][issue_index] 
+        print(color_sentence(sentences[r], issue_word)) 
```

Comment:
Instead of highlighting only one token per sentence, let's have an optional `token_score_threshold` arg for this function and additionally highlight any tokens whose score falls below the threshold (still highlighting the argmin always). The default value of the threshold can be quite low, say 1e-3.